core/test-output-stream.ts

```diff
@@ -106,12 +107,40 @@ export class TestOutputStream extends ReadableStream {
       return "undefined";
    }
 
-   private _getErrorYaml(error: MatchError): string {
-      return " ---\n" +
-      "   message: \"" + error.message.replace(/\\/g, "\\\\").replace(/"/g, "\\\"") + "\"\n" +
-      "   severity: fail\n" +
-      "   data:\n" +
-      "     got: " + JSON.stringify(error.actualValue) + "\n" +
-      "     expect: " + JSON.stringify(error.expectedValue) + "\n ...\n";
+   private _writeMatchErrorOutput(error: MatchError): void {
+
+       let sanitisedMessage = error.message.replace(/\\/g, "\\\\").replace(/"/g, "\\\"");
+       let sanitisedActual = JSON.stringify(error.actualValue);
+       let sanitisedExpected = JSON.stringify(error.expectedValue);
+
+       this._writeFailure(sanitisedMessage, sanitisedActual, sanitisedExpected);
+
+   }
+
+   private _writeUnhandledErrorOutput(error: Error): void {
+
+       this._writeFailure("The test threw an unhandled error.", "an unhandled error", "no unhandled errors to be thrown", error.stack);
```

Comment:
I think it's good to have a private method for it to keep it as separated as possible. The less logic in each function the better!